src/main/java/org/gradle/profiler/Main.java

```diff
@@ -74,27 +73,31 @@ public void run(String[] args) throws Exception {
                 } else {
                     throw new IllegalArgumentException("Don't know how to run scenario.");
                 }
-                Consumer<BuildInvocationResult> resultConsumer = benchmarkResults.scenario(scenario, invoker.samplesFor(settings));
                 try {
-                    invoker.run(scenario, settings, resultConsumer);
+                    invoker.run(scenario, settings, benchmarkResults);
                 } catch (Throwable t) {
                     t.printStackTrace();
                     failures.add(t);
                 } finally {
-                    // write the results up to this point
+                    // Write the current results and generate the reports, so that if this process crashes the results (which may have taken quite some time to collect) are not lost.
```

Comment:
Is this a common scenario?

Reply:
Maybe not any more. By "crash" I mean "some error that causes gradle-profiler to exit without writing the results". This happened to me a few times in a few different ways. For example, starting some long running benchmarks on a remote machine, forgetting to use `screen` or similar and then having the ssh connection die and the gradle-profiler JVM killed by the OS. Or when we messed up the build failure handling logic and the gradle-profiler JVM exited with an exception without writing the results instead of continuing with the next scenario. I can't say I ever saw the JVM crash, eg segfault, though.

It's just to add a little bit of robustness in cases where we don't deal with problems in a graceful way.